library/src/main/java/eu/kanade/tachiyomi/source/Source.kt

```diff
@@ -21,6 +23,36 @@ interface Source {
      */
     val name: String
 
+    /**
+     * Represents an IETF BCP 47 compliant language tag.
+     * Special cases include:
+     * - [Language.MULTI]: Indicates multiple languages.
+     * - [Language.OTHER]: Refers to a language not explicitly defined.
+     *
+     * Besides the above two [Language.ALL] (all) is also supported due to legacy reasons.
+     * Usage of it is highly discouraged
+     *
+     * @since extensions-lib 1.5
+     */
+    val language: String
+
+    /**
+     * Returns the list of filters for the source.
+     *
+     * @since extensions-lib 1.5
+     */
+    fun getFilterList(): FilterList
+
+    /**
+     * Get a page with a list of manga.
+     *
+     * @since extensions-lib 1.5
+     * @param query the search query.
+     * @param filters the list of filters to apply.
+     * @param page the page number to retrieve.
+     */
+    suspend fun getMangaList(query: String, filters: FilterList, page: Int): MangasPage = throw Exception("Stub!")
```

Comment:
why a new search method in `Source`? what about `getSearchManga`

Reply:
the PR is towards the `main` branch which doesn't have `getSearchManga` but either way this is basically replacing it.